File: Java/001_Two_Sum.java
/**
  Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
  You may assume that each input would have exactly one solution, and you may not use the same element twice.
  You can return the answer in any order.
 */

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> maps = new HashMap<>();
        for (int i = 0; i < nums.length; ++i){
            if (maps.containsKey(target - nums[i])){
                return new int[] {maps.get(target - nums[i]), i};
            } else{
                maps.put(nums[i], i);
            }
        }
        return new int[] {-1, -1};   
    }
}

File: JavaScript/001_Two_Sum.js
/**
 * Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.
 * You may assume that each input would have exactly one solution, and you may not use the same element twice.
 * You can return the answer in any order.
 */

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let numberMap = new Map();

    for (let index = 0; index < nums.length; index++) {
        val = nums[index];

    if (numberMap.has(target - val)) 
        return [index, numberMap.get(target - val)];
    else numberMap.set(val, index);
    }

    return [];
};